core/runtime/memory.hpp

```diff
@@ -70,35 +58,76 @@ namespace kagome::runtime {
     outcome::result<BytesOut> view(WasmSpan span) const {
       return view(PtrSize{span});
     }
+  };
+
+  /**
+   * A convenience wrapper around a memory handle and a memory allocator.
+   *
+   * Mind that the underlying memory can be accessed through unaligned pointers
+   * which isn't well-behaved in C++. WebAssembly nonetheless expects it to
+   * behave properly. Avoid emitting unaligned load/store by checking for
+   * alignment explicitly, and performing memcpy if unaligned.
+   *
+   * The allocated memory tries to have the same alignment as the memory being
+   * simulated.
+   */
+  class Memory final {
```

Comment:
Should we add TODO to remove this class?

Was it made to postpone refactoring of `memory./*allocate_and_*/write(bytes)` to `allocate_and_write(memory, allocator, bytes) = memory.write(allocator.allocate(memory, bytes.size), bytes)`?

`allocate_and_write` is only used in two places:
- module instance `call<Args>(Args)` wrapper method which writes arguments to wasm memory
- host api wrapper macro which writes returned value to wasm memory (we will refactor host api and extensions to return values instead of wasm pointer)

Reply:
It is a convenient wrapper for the current code. If we refactor something in the future, then we can consider removing this wrapper.